feat: add content about vmimage backup (#613)
harvester/docs: docs/vm/backup-restore.md

### Restore a new VM on another Harvester cluster

_Available as of v1.4.0_

Users can now restore a new VM on another cluster by leveraging the VM metadata & content backup feature.

> Harvester will handle VMImage data automatically. If a volume is based on a VMImage and there is a backup for it, Harvester will automatically sync the VMImage to backup target.

Reviewer: Since it's handled automatically, should we remove the section too?
```
#### Upload the same VM images to a new cluster
```

Reviewer: suggest changing `VMImage` to VM image for consistency across the document.

Author: Thanks for the suggestion. I addressed all comments.

### Restore a new VM on another Harvester cluster

_Available as of v1.4.0_

Users can now restore a new VM on another cluster by leveraging the VM metadata & content backup feature.

> Harvester will handle VM image data automatically. If a volume is based on a VM image and there is a backup for it, Harvester will automatically sync the VM image to backup target.

Reviewer 2: ```suggestion
Harvester automatically handles VM image data. If a volume is based on a VM image with a backup, Harvester automatically syncs the VM image with the backup target.
```

> In a new cluster, it the VMbackup is restored to a new VM, Harvester will automatically create the VM image from the backup target.

Reviewer 2: ```suggestion
In a new cluster, if the VM backup is restored to a new VM, Harvester automatically creates the VM image from the backup target.
```
